### PR TITLE
Attempt to fix #151

### LINKED ISSUE #151: Error: Cannot destructure property `edges` of 'undefined' or 'null'.
First off, thanks for the package!

This action has worked fine for me in the past but it has not been able to publish during some recent runs. It gives the error specified above. Here is the action [log](https://github.com/IgnisDa/printr/runs/2387331681?check_suite_focus=true#step:4:10).

EDIT:
Okay the error was raised because I forgot to change the version in `Cargo.toml` and cargo was trying to publish a version that had already been published. The error was fixed.

It would be nice to have some useful error messages for this, right?

### PR DESCRIPTION
**Pull request by katyo**

### CODE REVIEW

File: src/github.ts [hunk 1]
```diff
@@ -84,10 +84,25 @@ query lastCommitDate($owner: String!, $repo: String!, $ref: String!, $path: Stri
         )
     }
 
-    const {edges} = result.repository.ref.target.history
+    const {history} = result.repository.ref.target
```
Comment (wangfenjin) [inline src/github.ts:87]: I test the code, `result.repository.ref` this is null, so `result.repository.ref.target` throws `cannot read property 'target' of null`

**Comment by wangfenjin** [comment]
I think we can disable these logic, and just check the published version?

https://github.com/wangfenjin/publish-crates/blob/926eb3732ca6823d0876df86520f0f11e41c0cf2/src/main.ts#L58-L70

File: src/github.ts (hunk 1 above)
Comment (katyo) [reply src/github.ts:87]: Looks really strange. Seems Github's graphql API has unexpected behavior in some rare cases.

Comment (katyo) [reply src/github.ts]: Could you try again?

**Comment by wangfenjin** [comment]
https://github.com/wangfenjin/duckdb-rs/pull/23/checks?check_run_id=3575158281

File: src/github.ts (hunk 1 above)
Comment (wangfenjin) [reply src/github.ts]: Error: Unable to retrieve history for path 'libduckdb-sys' (res: {"repository":{"ref":null}})

**Comment by wangfenjin** [comment]
Seems related to inner package?

**Comment by katyo** [comment]
@wangfenjin Seems the problem related to `$GITHUB_REF` variable value.

API request:
```graphql
query lastCommitDate($owner: String!, $repo: String!,
                     $ref: String!, $path: String!) {
    repository(owner: $owner, name: $repo) {
        ref(qualifiedName: $ref) {
            target {
                ... on Commit {
                    history(first: 1, path: $path) {
                        edges {
                            node {
                                committedDate
                            }
                        }
                    }
                }
            }
        }
    }
}
```

Request data:
```json
{
  "owner": "wangfenjin",
  "repo": "duckdb-rs",
  "ref": "refs/remotes/pull/23/merge",
  "path": "libduckdb-sys"
}
```

Response from GitHub:
```json
{
  "data": {
    "repository": {
      "ref": null
    }
  }
}
```

We may try to find solution or bypass this logic as a quick fix.

**Comment by katyo** [comment]
@wangfenjin At all I think publishing packages using pull-requests is not so good idea in any case. I suggest you use tags for that purpose.

**Comment by katyo** [comment]
Seems we can get last path modification time locally via git.

**Comment by wangfenjin** [comment]
> @wangfenjin At all I think publishing packages using pull-requests is not so good idea in any case. I suggest you use tags for that purpose.

Agree. For that because it's the first time using github action to release, so I want to make it work before merge the code.

So you mean, after merge to master all things should work? What about the dry-run mode, is it work in PR? Because every PR should run the dry-run mode

**Comment by katyo** [comment]
@wangfenjin Currently I'm on a way to change implementation of this feature. New implementation will works with local clone of repository directly without using octokit. Also it will checks contents of packages more fine grained i.e. file by file.
Maybe I will release `v2` of this action. It shall works as expected in your case.

**Comment by katyo** [comment]
Closed in favor of #311

**Comment by katyo** [comment]
@wangfenjin Could you test pr #311 ?

**Pull request closed without merging**